### PR TITLE
Sync: Do not write PT username to server logs

### PR DESCRIPTION
**Pull request by marksvc**
- Note that the PutNotes test change may look nice, but that test
isn't working right now anyway :-/. See SF-1089.
- Introduces class MockLogger<T>, since using NSubstitute with ILogger
is tricky.

---
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/818)

### CODE REVIEW

**Comment by marksvc** [comment]
Oops. If I revise the PR, note to self to change the subject so it is a command. ("Do not write")

**Review by irahopkinson — CHANGES_REQUESTED** [review]
> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/818)* status: all files reviewed, 1 unresolved discussion (waiting on @marksvc)

___
*[test/SIL.XForge.Scripture.Tests/Services/ParatextServiceTests.cs, line 285 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/818#-MI3f8z18mIox8AbwmoA:-MI3f8z18mIox8AbwmoB:bk7rt8y) ([raw file](https://github.com/sillsdev/web-xforge/blob/97af8a3a8e958db6c86f49862e571605447ae115/test/SIL.XForge.Scripture.Tests/Services/ParatextServiceTests.cs#L285)):*
> ```
>  Assert.That(env.MockLogger.Messages.Any((string message) => message.Contains(env.Username01)), Is.False);
> ```
NSubsitute has a way to do this. Use it and remove the mock class. See https://nsubstitute.github.io/help/actions-with-arguments/ and the second example under **Performing actions with arguments** is probably exactly what you need.

**Review by marksvc — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/818)* status: all files reviewed, 1 unresolved discussion (waiting on @irahopkinson)

___
*[test/SIL.XForge.Scripture.Tests/Services/ParatextServiceTests.cs, line 285 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/818#-MI3f8z18mIox8AbwmoA:-MI4xcXh1TUBt8M78vAM:bk0ahr5) ([raw file](https://github.com/sillsdev/web-xforge/blob/97af8a3a8e958db6c86f49862e571605447ae115/test/SIL.XForge.Scripture.Tests/Services/ParatextServiceTests.cs#L285)):*
<details><summary><i>Previously, irahopkinson (Ira Hopkinson) wrote…</i></summary><blockquote>

> ```
>  Assert.That(env.MockLogger.Messages.Any((string message) => message.Contains(env.Username01)), Is.False);
> ```
NSubsitute has a way to do this. Use it and remove the mock class. See https://nsubstitute.github.io/help/actions-with-arguments/ and the second example under **Performing actions with arguments** is probably exactly what you need.
</blockquote></details>

I worked on using NSubstitute to mock ILogger<ParatextService> and was unsuccessful. The ILogger<> interface only has one logging method, but I was never able to catch any calls to Log(). I even had code that was exactly the same as a blog post showing how to do it. 

In other words, the MockLogger class wasn't made to be a simpler way of recording and allowing querying of the logged messages, rather than using NSubstitute. But rather because I wasn't able to get Substitute.For<ILogger<ParatextService>>().DidOrDidNotReceive.Log(arguments) to ever trip. 

If you are able to get a substituted ILogger<> to catch a .Log() call, maybe you can show me how?

**Review by marksvc — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/818)* status: all files reviewed, 1 unresolved discussion (waiting on @irahopkinson)

___
*[test/SIL.XForge.Scripture.Tests/Services/ParatextServiceTests.cs, line 285 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/818#-MI3f8z18mIox8AbwmoA:-MI50cmBEzTEncjEpgK2:b-cwa8nj) ([raw file](https://github.com/sillsdev/web-xforge/blob/97af8a3a8e958db6c86f49862e571605447ae115/test/SIL.XForge.Scripture.Tests/Services/ParatextServiceTests.cs#L285)):*
<details><summary><i>Previously, marksvc wrote…</i></summary><blockquote>

I worked on using NSubstitute to mock ILogger<ParatextService> and was unsuccessful. The ILogger<> interface only has one logging method, but I was never able to catch any calls to Log(). I even had code that was exactly the same as a blog post showing how to do it. 

In other words, the MockLogger class wasn't made to be a simpler way of recording and allowing querying of the logged messages, rather than using NSubstitute. But rather because I wasn't able to get Substitute.For<ILogger<ParatextService>>().DidOrDidNotReceive.Log(arguments) to ever trip. 

If you are able to get a substituted ILogger<> to catch a .Log() call, maybe you can show me how?
</blockquote></details>

I wrote some more in chat.

**Review by irahopkinson — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/818#-:-MIr4fzp-yTl0pwhsQ2H:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved

___
*[test/SIL.XForge.Scripture.Tests/Services/ParatextServiceTests.cs, line 285 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/818#-MI3f8z18mIox8AbwmoA:-MIr4VIoAPug1gVOrqIu:b-rbv6h1) ([raw file](https://github.com/sillsdev/web-xforge/blob/97af8a3a8e958db6c86f49862e571605447ae115/test/SIL.XForge.Scripture.Tests/Services/ParatextServiceTests.cs#L285)):*
<details><summary><i>Previously, marksvc wrote…</i></summary><blockquote>

I wrote some more in chat.
</blockquote></details>

I gave it a go and with @ddaspit help I think I understand the issue at hand now. Your solution is the right option for this.
